ddtrace/tracer/textmap.go

```diff
@@ -501,6 +504,97 @@ func (*propagatorB3) extractTextMap(reader TextMapReader) (ddtrace.SpanContext,
 	return &ctx, nil
 }
 
+// propagatorB3 implements Propagator and injects/extracts span contexts
+// using B3 headers. Only TextMap carriers are supported.
+type propagatorB3SingleHeader struct{}
+
+func (p *propagatorB3SingleHeader) Inject(spanCtx ddtrace.SpanContext, carrier interface{}) error {
+	switch c := carrier.(type) {
+	case TextMapWriter:
+		return p.injectTextMap(spanCtx, c)
+	default:
+		return ErrInvalidCarrier
+	}
+}
+
+func (*propagatorB3SingleHeader) injectTextMap(spanCtx ddtrace.SpanContext, writer TextMapWriter) error {
+	ctx, ok := spanCtx.(*spanContext)
+	if !ok || ctx.traceID == 0 || ctx.spanID == 0 {
+		return ErrInvalidSpanContext
+	}
+	sb := strings.Builder{}
+	sb.WriteString(fmt.Sprintf("%016x-%016x", ctx.traceID, ctx.spanID))
+	if p, ok := ctx.samplingPriority(); ok {
+		if p >= ext.PriorityAutoKeep {
+			sb.WriteString("-1")
+		} else {
+			sb.WriteString("-0")
+		}
+	}
+	writer.Set(b3SingleHeader, sb.String())
+	return nil
+}
+
+func (p *propagatorB3SingleHeader) Extract(carrier interface{}) (ddtrace.SpanContext, error) {
+	switch c := carrier.(type) {
+	case TextMapReader:
+		return p.extractTextMap(c)
+	default:
+		return nil, ErrInvalidCarrier
+	}
+}
+
+func (*propagatorB3SingleHeader) extractTextMap(reader TextMapReader) (ddtrace.SpanContext, error) {
+	var ctx spanContext
+	err := reader.ForeachKey(func(k, v string) error {
+		var err error
+		key := strings.ToLower(k)
+		switch key {
+		case b3SingleHeader:
+			b3Parts := strings.Split(v, "-")
+			if len(b3Parts) >= 2 {
+				if len(b3Parts[0]) > 16 {
+					b3Parts[0] = b3Parts[0][len(b3Parts[0])-16:]
+				}
+				ctx.traceID, err = strconv.ParseUint(b3Parts[0], 16, 64)
+				if err != nil {
+					return ErrSpanContextCorrupted
+				}
+
+				ctx.spanID, err = strconv.ParseUint(b3Parts[1], 16, 64)
+				if err != nil {
+					return ErrSpanContextCorrupted
+				}
+				if len(b3Parts) >= 3 {
+					if b3Parts[2] != "" {
+						switch b3Parts[2] {
+						case "1":
+							ctx.setSamplingPriority(1, samplernames.Unknown)
+						case "0":
+							ctx.setSamplingPriority(0, samplernames.Unknown)
+						case "d": // Treat 'debug' traces as priority 1
+							ctx.setSamplingPriority(1, samplernames.Unknown)
+						default:
+							return ErrSpanContextCorrupted
+						}
+					}
+				}
```

Comment:
```suggestion
				if len(b3Parts) >= 3 {
					if b3Parts[2] != "" {
						switch b3Parts[2] {
						case "1", "d": // Treat 'debug' traces as priority 1
							ctx.setSamplingPriority(1, samplernames.Unknown)
						case "0":
							ctx.setSamplingPriority(0, samplernames.Unknown)
						default:
							return ErrSpanContextCorrupted
						}
					}
				}
```

Reply:
👍 I've updated the code to almost exactly this. Unfortunately I need to keep a separate `case ""` as not providing a priority is still a valid header